src/fenicsxconcrete/experiment.py

```diff
@@ -0,0 +1,35 @@
+from pydantic import RootModel
+from bcs import DirichletBCDefinition, InitialConditionDefinition, NeumannBCDefinition, BodyForceDefinition
+from mesh import MeshGenerator
+from material import MaterialDefinition, LinearElasticMaterial
+from pydantic.dataclasses import dataclass
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class Experiment:
+    dirichlet_bcs: list[DirichletBCDefinition] | None
+    neumann_bcs: list[NeumannBCDefinition] | None
+    initial_conditions: list[InitialConditionDefinition] | None
+    body_forces: list[BodyForceDefinition] | None
+    geometry: MeshGenerator
+    solution_fields: list[str]
+    time: tuple[float, float]
+    material: MaterialDefinition
+    name: str = "experiment"
+
+
+
+
+if __name__ == "__main__":
+    bc = DirichletBCDefinition(marker=1, value=[0.,0.], subspace=0, variable='displacement')
+    neumann = NeumannBCDefinition(marker=2, value=[-42.0], variable='displacement')
+    initial = InitialConditionDefinition(value=[42.24], variable='density')
+    body_force = BodyForceDefinition(value=[0.,0., 9.81], variable='displacement')
+    mat = LinearElasticMaterial(name='steel', mu=1., lam=2.)
+    geo = MeshGenerator(parameters={'length': (1, 'm')}, mesh_tags={'left': 0, 'right': 1, 'top': 2, 'bottom': 3})
```

Comment:
Not sure yet. Could be a path to an xdmf or a `dolfinx.mesh.Mesh`. We could also just decide that a mesh-generator always produces a `.msh` file.

Reply:
I'm not the expert here, but it would be important to get a general agreement.

Reply:
Parallel input is supported by xdmf, but `dolfinx.io.gmshio` will read the file on a single process.
If `MeshGenerator` uses `Gmsh` under the hood and also creates physical groups etc. then always producing a `.msh` and providing methods to convert to xdmf would be my personal choice.
(This may also be useful for debugging of facet tags etc.)

Reply:
How can we cover the case that the mesh is given from other software (abaqus ...) or someone send me the mesh fiel and I have to use that? MeshGenerator = reading existing file and defining boundary groups (this would then be fenics stuff, or?)

Reply:
We could have multiple meshGenerators, one creating basic meshes, another one that reads in from any specific file format. For the bc, we could either use the physical components from these mesh files, or create these components later by using functional representations as in fenics.